code/game/turfs/open/lava.dm

```diff
@@ -166,6 +166,8 @@
 
 				if(S && H && S.clothing_flags & LAVAPROTECT && H.clothing_flags & LAVAPROTECT)
 					return
+				else
+					C.update_fire(force = TRUE)//you will always be burning on top of lava if you're not protected
```

Comment:
There's already update_fire above, and there's another return below so this seems like inappropriate location ?

Reply:
hummm...